pkg/resource/deploy/source_eval.go

```diff
@@ -214,6 +217,8 @@ func (iter *evalSourceIterator) Next() (SourceEvent, error) {
 
 	// Await the program to compute some more state and then inspect what it has to say.
 	select {
+	case <-iter.mon.AbortChan():
+		return nil, result.BailErrorf("EvalSourceIterator aborted")
```

Comment:
I feel like this is either or, maybe -- you either leave the type in the struct be `SourceResourceMonitor`, and lean on an `AbortChan` interface method here, or you use `*resmon` in the struct and the `abortChan` field directly. Thoughts? Apologies if I've missed something!

Reply:
Yep my bad :man_facepalming: